### ISSUE TITLE
`UniTask.Yield()` continues to run for 2 frames after the Unity editor is stopped

### ISSUE THREAD

**Issue by develax** [Cysharp/UniTask#305]
Here is a test component:

```
using Cysharp.Threading.Tasks;
using UnityEditor;
using UnityEngine;

public class UniTaskTest : MonoBehaviour
{
    private void Start()
    {
        testAsync();
    }

    private async void testAsync()
    {
        while (true)
        {
            await UniTask.Yield();
            Debug.Log($"UniTask.Yield: {Time.frameCount}, isPlaying: {EditorApplication.isPlaying}");
        }
    }

    private void Update()
    {
        Debug.Log($"Update: {Time.frameCount}, isPlaying: {EditorApplication.isPlaying}");
    }

    private void OnDestroy()
    {
        Debug.Log($"OnDestroy: {Time.frameCount}, isPlaying: {EditorApplication.isPlaying}");
    }
}
```

Console output:
![image](https://user-images.githubusercontent.com/13222593/126905638-3a32a435-7a75-48b5-a283-47d4b3ebaf60.png)

I know there is `cancellationToken: this.GetCancellationTokenOnDestroy()` to prevent this but it is not exactly the same as checking `EditorApplication.isPlaying` since it has 2 flaws (IMHO):

1. `GetCancellationTokenOnDestroy` results in an error in the console
2. necessity to write extra code for the standard behavior.

Would it be possible to implement it with `EditorApplication.isPlaying == true` as the default behavior?
Or maybe there are some reasons for this behavior by design?

**Comment by develax** [comment]
As a temp solution I added this "dirty" hack to the `ContinuationQueue` class:

![image](https://user-images.githubusercontent.com/13222593/126907668-6aa9247b-dc49-4258-9da5-6a8c38f04c8f.png)

It works for me, but I'm not sure if it's safe and doesn't break anything.

**Comment by neuecc** [comment]
recommend to use `GetCancellationTokenOnDestroy`, always if you want to care.

> GetCancellationTokenOnDestroy results in an error in the console

No.
When managed by UniTask, OperationCancelationException will not be logged in console.
`private async void testAsync()` is invalid, it is managed by standard Task.
`private async UniTaskVoid testAsync()` is valid.

**Comment by develax** [comment]
`UniTaskVoid` helped with the console error, thanks!

Still have 2 question:
1. If `void` is managed by standard `Task` why my intervention in the `ContinuationQueue` helped?
2. Why checking for `EditorApplication.isPlaying` (and stopping when equals `false`) can't be implemented as default behaviour? Who needs this behaviour when it runs 2 frames longer when the editor is stopped? IMHO this only provokes unexpected issues. For example, if some `async` code performs changes to the scene game-objects after the editor is stopped, these changes will stay in the scene and the scene won't be even marked as changed in the inspector window, therefore these changes will most likely not be noticed by the developer and will go into production. Isn't it good when the library is designed to prevent non-obvious developer errors?

Just want to understand how it works and tech reasons.

**Comment by neuecc** [comment]
1. UniTask.Yield is inside, running on UniTask(ContinuationQueue). `async void` is outside.

2. The exact place where controlling is not there, but here.
https://github.com/Cysharp/UniTask/blob/60bfbae787b1aade3b7a2504999031d96996d7dd/src/UniTask/Assets/Plugins/UniTask/Runtime/PlayerLoopHelper.cs#L202-L222
// run rest action before clear.
is the intended behavior.
The intention is to check the last CancellationToken before the forced queue clear to prevent illegal control on UnityEditor.
This is useless if you are not using a CancellationToken.
In other words, please use the CancellationToken.
